mne/channels/channels.py

```diff
@@ -387,6 +390,11 @@ def set_channel_types(self, mapping):
             else:
                 coil_type = FIFF.FIFFV_COIL_NONE
             self.info['chs'][c_ind]['coil_type'] = coil_type
+        if len(list(unit_changes.keys())) > 0:
+            msg = "The unit for channel(s) {0} has changed from {1} to {2}."
+            for this_change in unit_changes:
+                names = ", ".join(sorted(unit_changes[this_change]))
+                warn(msg.format(names, *this_change))
```

Comment:
I would get rid of the if and just do ``for key, value in unit_changes.iteritems():``.